pymc3/tests/test_distributions_random.py

```diff
@@ -491,43 +414,364 @@ class TestMoyal(BaseTestCases.BaseTestCase):
     params = {"mu": 0.0, "sigma": 1.0}
 
 
-@pytest.mark.skip(reason="This test is covered by Aesara")
-class TestCategorical(BaseTestCases.BaseTestCase):
-    distribution = pm.Categorical
-    params = {"p": np.ones(BaseTestCases.BaseTestCase.shape)}
+class BaseTestDistribution(SeededTest):
+    pymc_dist: Optional[Callable] = None
+    pymc_dist_params = dict()
+    reference_dist: Optional[Callable] = None
+    reference_dist_params = dict()
+    expected_rv_op_params = dict()
+    tests_to_run = []
+    size = 15
+    decimal = 6
+
+    sizes_to_check: Optional[List] = None
+    sizes_expected: Optional[List] = None
+    repeated_params_shape = 5
+
+    def test_distribution(self):
+        self._instantiate_pymc_rv()
+        if self.reference_dist is not None:
+            self.reference_dist_draws = self.reference_dist()(
+                **self.reference_dist_params, size=self.size
+            )
+        for test_name in self.tests_to_run:
+            self.run_test(test_name)
+
+    def run_test(self, test_name):
```

Comment:
This can just be a class dictionary with name:class methods, no?

Reply:
I think that it may result uglier than how it is at the moment. 

I would need to convert all the `_check` methods in class methods and have the class dictionary being defined after these, so that it could refer to them. But also the class dictionary will need to be defined right before `test_distribution`, so that this could also refer to it. As per below
```python
class BaseTestDistribution:

    @classmethod
    def _check_1(cls):
        assert True
    
    avaliable_tests = {
        "check_1": _check_1
    }
    
    def test_distributions(self):
        self.avaliable_tests["check_1"]()

```
If we don't want to re-define the dictionary for each single test-class, or if we want to get rid of unnecessary code and make it a bit more implicit we could simply do a bit of renaming and something along these lines:
```python
def run_test(self, test_name):
    getattr(self, test_name)()
````

Reply:
Thanks for the explanation. I agree with your point. I think the last approach is nicer than the current one, and you can drop the `run_test` function all together in that case as well, no?

```python
for test_name in self.tests_to_run:
  getattr(self, test_name)()
```